### PR TITLE
Tcs 30 data store to subscribe to particular segmented dataset

### PR DESCRIPTION
**Pull request by CReid98**
- creating a new segmentedsubdetails function to subscribe to striped segmented tickerplant
- subscribe function will retrieve filters from stp based on segment id defined in config file

### CODE REVIEW

File: code/processes/segmentedtickerplant.q [hunk 1]
```diff
@@ -71,6 +76,17 @@ setup:{[batch]
...
+loadsubfilters:{[]
+	.sub.filterparams:@[{1!("S**";enlist",")0: x};.rdb.subcsv;{.lg.e[`loadsubfilters;"Failed to load .rdb.subcsv with error: ",x]}];
+	.rdb.subscribeto:raze value flip key .sub.filterparams;
+	.rdb.subscribesyms:.sub.filterparams;}
+
+// load config file if it exists
+// just giving some sample config
+tpconfig: ([] table:`trade`trade`quote`quote; segmentID:1 2 1 2; wcRef:`wcA`wcB`wcC`wcD; filters:`AAPL`GOOG`AAPL`GOOG)
+procmap: (enlist `rdb1)!(enlist 1)
+
```
Comment (rian-morgan) [inline code/processes/segmentedtickerplant.q:89]: As above, all this needs to be removed. Be careful about what you commit. This is all temp code and not required.

File: code/rdb/datastripe.q [hunk 2]
```diff
@@ -0,0 +1,3 @@
+\d .ds
+
+segmentid: "J"$@[{(.Q.opt .z.x)[x]};`segid;0n]
```
Comment (rian-morgan) [inline code/rdb/datastripe.q:3]: add some descriptive comment after this plz.

File: code/common/subscriptions.q [hunk 3]
```diff
@@ -129,7 +129,7 @@ subscribe:{[tabs;instrs;setschema;replaylog;proc]
     d:(`subtables`tplogdate!(details[`schemalist][;0];(first "D" $ -10 sublist string last first details[`logfilelist])^logdate));
     :d,{(where 101 = type each x)_x}(`i`icounts`d)!(details[`logfilelist][0;0];details[`rowcounts];details[`date])];
   if[tptype~`segmented;
-    retdic:`logdir`subtables!(details[`logdir];details[`schemalist][;0]);
+    retdic:`logdir`subtables`tptype!(details[`logdir];details[`schemalist][;0];`segmented);
```
[the remark this answers is not in the document]
Comment (CReid98) [reply code/common/subscriptions.q:132]: This was also meant to be temporary code, I have removed it

File: code/processes/segmentedtickerplant.q [hunk 4]
```diff
@@ -31,6 +31,11 @@ subdetails:{[tabs;instruments]
  `schemalist`logfilelist`rowcounts`date`logdir!(.ps.subscribe\:[tabs;instruments];.stplg.replaylog[tabs];tabs#.stplg `rowcount;(.eodtime `d);`$getenv`KDBTPLOG)
  }
 
+// subdetails function adapted for segmented tickerplant 
+/segmentedsubdetails:{[tabs;instruments;id]
+/ `schemalist`logfilelist`rowcounts`date`logdir`filters!(.ps.subscribe\:[tabs;instruments];.stplg.replaylog[tabs];tabs#.stplg `rowcount;(.eodtime `d);`$getenv`KDBTPLOG;[select table,wcRef,filters from tpconfig where segmentID=procmap[id]])
+/ }
+
```
[the remark this answers is not in the document]
Comment (CReid98) [reply code/processes/segmentedtickerplant.q:38]: removed

File: code/processes/segmentedtickerplant.q (hunk 1 above)
Comment (CReid98) [reply code/processes/segmentedtickerplant.q:89]: removing this, will also look at error trapping for when there is no config

File: code/rdb/datastripe.q (hunk 2 above)
Comment (CReid98) [reply code/rdb/datastripe.q:3]: added comment

File: code/wdb/datastripe.q [hunk 5]
```diff
@@ -0,0 +1,3 @@
+\d .ds
+
+segmentid: "J"$@[{(.Q.opt .z.x)[x]};`segid;0n]
```
[the remark this answers is not in the document]
Comment (CReid98) [reply code/wdb/datastripe.q:3]: added comment

**Review by rian-morgan — CHANGES_REQUESTED** [review]

**Review by rian-morgan — APPROVED** [review]
looks good

**Comment by rian-morgan** [comment]
https://github.com/AquaQAnalytics/TorQ/pull/414#pullrequestreview-890064267

**Review by jwrno1 — APPROVED** [review]

**Pull request merged**